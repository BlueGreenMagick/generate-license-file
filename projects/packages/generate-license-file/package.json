{
  "name": "generate-license-file",
  "version": "0.0.0",
  "description": "Generates a text file containing all of the licences for your production dependencies",
  "main": "dist/main.js",
  "license": "ISC",
  "prettier": "@generate-license-file/prettier-config",
  "bin": {
    "generate-license-file": "bin/generate-license-file"
  },
  "scripts": {
    "build": "del-cli ./dist&&tsc",
    "build:watch": "del-cli ./dist&&tsc --watch",
    "test": "jest",
    "test:ci": "jest --ci --coverage"
  },
  "contributors": [
    {
      "name": "Toby Smith",
      "url": "https://github.com/tobysmith568"
    },
    {
      "name": "Toby Bessant",
      "url": "https://github.com/tobybessant"
    }
  ],
  "files": [
    "dist",
    "bin"
  ],
  "keywords": [
    "cli",
    "license",
    "licence",
    "third",
    "party",
    "file",
    "generate",
    "oss"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/TobyAndToby/generate-license-file.git"
  },
  "bugs": {
    "url": "https://github.com/TobyAndToby/generate-license-file/issues"
  },
  "homepage": "https://github.com/TobyAndToby/generate-license-file#readme",
  "dependencies": {
    "arg": "^5.0.0",
    "cli-spinners": "^2.6.0",
    "enquirer": "^2.3.6",
    "esm": "^3.2.25",
    "glob": "^8.0.0",
    "inquirer": "^9.0.0",
    "license-checker": "^25.0.1",
    "ora": "^5.4.1"
  },
  "devDependencies": {
    "@types/glob": "^7.1.4",
    "@types/inquirer": "^8.0.0",
    "@types/jest": "^28.0.0",
    "@types/license-checker": "^25.0.3",
    "del-cli": "^4.0.0"
  }
}
