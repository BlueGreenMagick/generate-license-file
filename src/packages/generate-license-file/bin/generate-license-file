#!/usr/bin/env node

require = require('esm')(module /*, options*/);
require('../src/lib/cli').main(process.argv);
