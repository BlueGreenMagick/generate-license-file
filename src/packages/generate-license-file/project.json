{
  "name": "generate-license-file",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/generate-license-file/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/generate-license-file",
        "main": "packages/generate-license-file/src/index.ts",
        "tsConfig": "packages/generate-license-file/tsconfig.lib.json",
        "assets": [
          "packages/generate-license-file/*.md",
          "packages/generate-license-file/bin/**",
          "LICENSE.md"
        ],
        "buildableProjectDepsInPackageJsonType": "dependencies"
      }
    },
    "version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/version.mjs generate-license-file {args.version}"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs generate-license-file {args.tag}"
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/generate-license-file/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "packages/generate-license-file/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
