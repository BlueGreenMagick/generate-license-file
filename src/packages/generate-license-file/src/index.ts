/* istanbul ignore file */
export { generateLicenseFile } from "./lib/generateLicenseFile";
export { getLicenseFileText } from "./lib/getLicenseFileText";
export { getProjectLicenses } from "./lib/getProjectLicenses";
export { LineEnding } from "./lib/lineEndings";
export { ILicense } from "./lib/models/license";
