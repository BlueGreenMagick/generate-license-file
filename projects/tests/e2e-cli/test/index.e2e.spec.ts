import { describeEachLineEnding, describeEachTestPackage } from "@generate-license-file/e2e-shared";
import { exec } from "child_process";
import fs from "fs/promises";
import { allLineEndings, LineEnding } from "generate-license-file";
import path from "path";
import { promisify } from "util";

const execAsync = promisify(exec);

describe("cli", () => {
  describeEachTestPackage(packageJsonUnderTest => {
    let outputFileName = "";

    let input = "";
    let output = "";

    beforeEach(async () => {
      outputFileName = `test-${Math.floor(Math.random() * 1000)}.txt`;

      input = packageJsonUnderTest;
      output = path.dirname(packageJsonUnderTest) + "/" + outputFileName;
    });

    afterEach(async () => {
      try {
        await fs.unlink(output);
      } catch {
        // empty
      }
    });

    it("should match snapshot with just --input and --output", async () => {
      await execAsync(`npx generate-license-file --input ${input} --output ${output}`);

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    it("should match snapshot with --output given before --input", async () => {
      await execAsync(`npx generate-license-file --output ${output} --input ${input}`);

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    it("should match snapshot using -i and -o", async () => {
      await execAsync(`npx generate-license-file -i ${input} -o ${output}`);

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    it("should match snapshot with --input and --output wrapped in double quotes", async () => {
      await execAsync(`npx generate-license-file --input "${input}" --output "${output}"`);

      const result = await fs.readFile(output, "utf8");
      expect(result).toMatchSnapshot();
    });

    describe("overwrite", () => {
      it("should overwrite the file if --overwrite is given", async () => {
        const fileContent = "File content to be overwritten";

        await fs.writeFile(output, fileContent);

        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --overwrite`,
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).not.toMatch(fileContent);
      });
    });

    describeEachLineEnding((lineEnding, lineEndingLiteral) => {
      let lineEndingsNotUnderTest: LineEnding[] = [];

      beforeEach(async () => {
        lineEndingsNotUnderTest = allLineEndings.filter(x => x !== lineEnding);
      });

      it("should match snapshot when --eol is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --eol ${lineEnding}`,
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });

      it("should contain the correct line ending value", async () => {
        const expectedLineEndingValue = lineEndingLiteral;

        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --eol ${lineEnding}`,
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toContain(expectedLineEndingValue);
      });

      lineEndingsNotUnderTest.forEach(otherLineEnding =>
        it(`should not contain the incorrect line ending value (${otherLineEnding})`, async () => {
          const incorrectLineEndingValue = lineEndingLiteral;

          await execAsync(
            `npx generate-license-file --input ${input} --output ${output} --eol ${lineEnding}`,
          );

          const result = await fs.readFile(output, "utf8");
          expect(result).not.toContain(incorrectLineEndingValue);
        }),
      );
    });

    describe("no-spinner", () => {
      it("should match snapshot when --no-spinner is given", async () => {
        await execAsync(
          `npx generate-license-file --input ${input} --output ${output} --no-spinner`,
        );

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });
    });

    describe("ci", () => {
      it("should match snapshot when --ci is given", async () => {
        await execAsync(`npx generate-license-file --input ${input} --output ${output} --ci`);

        const result = await fs.readFile(output, "utf8");
        expect(result).toMatchSnapshot();
      });
    });

    describe("version", () => {
      it("should not produce an output when --version is given", async () => {
        await execAsync(`npx generate-license-file --input ${input} --output ${output} --version`);

        await expect(fs.stat(output)).rejects.toThrow();
      });

      it("should not produce an output when -v is given", async () => {
        await execAsync(`npx generate-license-file -i ${input} -o ${output} -v`);

        await expect(fs.stat(output)).rejects.toThrow();
      });

      it("should return the current version when --version is given", async () => {
        const { stdout } = await execAsync(`npx generate-license-file --version`);

        expect(stdout.trim()).toBe("v0.0.0");
      });

      it("should return the current version when -v is given", async () => {
        const { stdout } = await execAsync(`npx generate-license-file -v`);

        expect(stdout.trim()).toBe("v0.0.0");
      });
    });
  });
});
