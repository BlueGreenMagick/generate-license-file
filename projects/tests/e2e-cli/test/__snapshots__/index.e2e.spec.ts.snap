// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli for package project-1/package.json with a relative path ci should match snapshot when --ci is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path for line ending crlf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path for line ending lf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path no-spinner should match snapshot when --no-spinner is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path should match snapshot using -i and -o 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path should match snapshot with --input and --output wrapped in double quotes 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path should match snapshot with --output given before --input 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with a relative path should match snapshot with just --input and --output 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path ci should match snapshot when --ci is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path for line ending crlf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path for line ending lf should match snapshot when --eol is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path no-spinner should match snapshot when --no-spinner is given 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path should match snapshot using -i and -o 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path should match snapshot with --input and --output wrapped in double quotes 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path should match snapshot with --output given before --input 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;

exports[`cli for package project-1/package.json with an absolute path should match snapshot with just --input and --output 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-five@1.0.0

This package contains the following license and notice below:

# Dep Five

| Test Case | NO LICENSE FILE  |
| --------- | ---------------- |
| Found     | TRUE (README.MD) |

## Notes

This README should be used as a substitute for the license file.

-----------

The following npm package may be included in this product:

 - dep-four@1.0.0

This package contains the following license and notice below:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one@1.0.0

This package contains the following license and notice below:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three@1.0.0

This package contains the following license and notice below:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two-duplicate@1.0.0
 - dep-two@1.0.0

These packages each contain the following license and notice below:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;
